config/yaml_test.go

```diff
@@ -255,9 +255,46 @@ func TestTypeOfTypes(t *testing.T) {
 		tts := typeStructStruct{}
 		err := provider.Get(Root).PopulateStruct(&tts)
 		assert.NoError(t, err)
+		assert.Equal(t, userDefinedTypeInt(123), tts.TypeStruct.TestInt)
+		assert.Equal(t, userDefinedTypeUInt(456), tts.TypeStruct.TestUInt)
+		assert.Equal(t, userDefinedTypeFloat(123.456), tts.TypeStruct.TestFloat)
+		assert.Equal(t, userDefinedTypeBool(true), tts.TypeStruct.TestBool)
+		assert.Equal(t, userDefinedTypeString("hello"), tts.TypeStruct.TestString)
+		assert.Equal(t, 10*time.Second, tts.TypeStruct.TestDuration.Seconds)
+		assert.Equal(t, 20*time.Minute, tts.TypeStruct.TestDuration.Minutes)
+		assert.Equal(t, 30*time.Hour, tts.TypeStruct.TestDuration.Hours)
+	})
+}
+
+func TestTypeOfTypesPtr(t *testing.T) {
+	withYamlBytes(t, typeStructYaml, func(provider Provider) {
+		tts := typeStructStructPtr{}
```

Comment:
I am not adding test for code I didn't write here. I am happy open a separate issue for that, otherwise this conversation will go on and on untill we cover all combinatorial of test cases..